src/main/webapp/app/exercises/shared/team/teams-import-dialog/teams-import-from-file-form.component.ts

```diff
@@ -0,0 +1,98 @@
+import { User } from 'app/core/user/user.model';
+import { Team, TeamList } from 'app/entities/team.model';
+
+@Component({
+    selector: 'jhi-teams-import-from-file-form',
+    templateUrl: './teams-import-from-file-form.component.html',
+    styleUrls: ['./teams-import-from-file-form.component.scss'],
+})
+export class TeamsImportFromFileFormComponent implements OnInit {
+    @Output() teamsChanged = new EventEmitter<Team[]>();
+    sourceTeams?: Team[];
+    importedTeams: TeamList = { students: [] };
+    importFile?: Blob;
+    importFileName: string;
+    loading: boolean;
+
+    constructor(private changeDetector: ChangeDetectorRef) {}
+
+    ngOnInit(): void {}
+
+    /**
+     * Move file reader creation to separate function to be able to mock
+     * https://fromanegg.com/post/2015/04/22/easy-testing-of-code-involving-native-methods-in-javascript/
+     */
+    generateFileReader() {
+        return new FileReader();
+    }
+
+    /**
+     * Converts teams from file to expected team type
+     * @param {FileReader} $fileReader object that is generated by generateFileReader
+     */
+    onFileLoadImport(fileReader: FileReader) {
+        try {
+            // Read the file and get list of questions from the file
+            this.importedTeams = JSON.parse(fileReader.result as string) as TeamList;
+            this.sourceTeams = this.convertTeams(this.importedTeams);
+            this.teamsChanged.emit(this.sourceTeams);
+            this.loading = false;
+            // this.verifyAndImportQuestions(questions);
+            // Clearing html elements,
+            this.importFile = undefined;
+            this.importFileName = '';
+            const control = document.getElementById('importFileInput') as HTMLInputElement;
+            if (control) {
+                control.value = '';
+            }
+        } catch (e) {
+            alert(`Import Teams Failed! Invalid teams file. ${e}`);
+        }
+    }
+
+    /**
+     * Assigns the uploaded import file
+     * @param $event object containing the uploaded file
+     */
+    setImportFile($event: any): void {
+        if ($event.target.files.length) {
+            const fileList: FileList = $event.target.files;
+            this.importFile = fileList[0];
+            this.importFileName = this.importFile['name'];
+            this.loading = true;
+        }
+        if (!this.importFile) {
+            return;
+        }
+        const fileReader = this.generateFileReader();
+        fileReader.onload = () => this.onFileLoadImport(fileReader);
+        fileReader.readAsText(this.importFile);
+        this.changeDetector.detectChanges();
+    }
+
+    /**
+     * Convert imported team list to normal teams
+     */
+    convertTeams(importTeam: TeamList): Team[] {
+        const teams: Team[] = [];
+        importTeam.students!.forEach((student) => {
+            const index = teams.findIndex((team) => team.name === student['Team Name']);
```

Comment:
Small remark: It would make the code a little bit more readable if you moved this line to line 85 because that's where you're using `index` for the first time.